app/controllers/shipit/ccmenu_controller.rb

```diff
@@ -0,0 +1,45 @@
+module Shipit
+  class CcmenuController < ActionController::Base
+    before_action :verify_signature, :check_permission
+
+    class NoDeploy
+      def id
+        0
+      end
+
+      def ended_at
+        DateTime.now.utc
+      end
+
+      def status
+        false
+      end
+
+      alias_method :running?, :status
+      alias_method :failed?, :status
+      alias_method :error?, :status
+      alias_method :success?, :status
+    end
+
+    def show
+      latest_deploy = stack.deploys[0] || NoDeploy.new
```

Comment:
`stack.deploy[0]` -> `stack.deploy.first`.

The former will load every single records in memory, and then call `Array#[]`. The later will add a `LIMIT 1` to the query and load a single record.